### PR TITLE
Basic test-infra

### LINKED ISSUE #3: Build a test framework
According to @you06 , cases in tipocket are managed manually, a typical work flow is:

user create the cluster to test -> run test binary (tipocket) -> witness and check -> user teardown cluster

Obviously, the lifecycle and configuration of cluster is not managed in the case.  I propose to adopt ginkgo as the test framework and abstract common operations as lib, e.g. provision cluster, apply chaos.

Ideal work flow after this issue get resolved (persudo code):

```go
ginkgo.BeforeEach(func() {
	chaos := NewChaosClient()
        tidbOps := NewTiDBOperationInterface()
})

ginkgo.Context("CDC: CDC should tolerant xxx chaos ", func() {
	instance := tidbOps.DeployCdcCluster(&CdcCluster{
               Source: &TiDBCluster{},
               Target: &MySQLTarget{},
         })
        chaos.PodKill(instance, &Config{Interval: "60s"})
        go func() {
              // case actions
        }()
        go func() {
             // case verifications
        }()
}
```                                          

Finally run with:

```
ginkgo --untilItFails --focus "CDC"
```

### PR DESCRIPTION
**Pull request by aylei**
close #3 

There are some preconditions for this PR to work:

- ~[x] Fix https://github.com/pingcap/tidb-operator/issues/1121~ current workaround is to implement this in `tipocket` first
- ~[x] Fix the CDC client dependency issue~ current workaround is to call the cdc binary

The framework borrowed from kubernetes is generally workable, but there is still a problem in `WaitForTiDBClusterReady` method, I prefer to merge this first and fix that method in another PR. Because actually the way to create TiDB cluster is not relevant to the framework.

case example:

https://github.com/pingcap/tipocket/pull/5/files#diff-1fab781b93dc2f9d325ba1eb4833fc02

Signed-off-by: Aylei <rayingecho@gmail.com>

### CODE REVIEW

**Comment by aylei** [comment]
@you06 Could you grant me the privilege to request review?

**Comment by you06** [comment]
@aylei done.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Tipocket
+
+Run cases locally:
+
+```bash
+> KUBECONIFG=<YOU_KUBECONFIG_PATH> make test
```
Comment (you06) [inline README.md]:
```suggestion
> KUBECONFIG=<YOU_KUBECONFIG_PATH> make test
```

**Comment by you06** [comment]
Besides, the cluster and chaos plan can be described by a config file either some BDD style code, I'm not sure which one is better.

**Comment by mahjonp** [comment]
> Besides, the cluster and chaos plan can be described by a config file either some BDD style code, I'm not sure which one is better.

I think BDD style case is easy to test and debug, it resolve one of main pain points of Schrodinger which is hard to debug a new case locally. what do you think? @you06

**Comment by aylei** [comment]
@you06 @mahjonp I prefer BDD-style code, but It takes a lot of effort to write a handy BDD interface.

**Pull request merged**